docs/cdms/cdms.rst

```diff
@@ -0,0 +1,237 @@
+   28503  CO, v=0     95      2.5595     2.2584     2.0369     1.9123     1.7370    1.4386      1.1429       0.8526       0.5733       0.3389       0.1478
+
+
+One of the advantages of using CDMS is the availability in the catalog of the
+partition function at different temperatures for the molecules (just like for
+JPL). As a continuation of the example above, an example that accesses and
+plots the partition function against the temperatures found in the metadata is
+shown below:
+
+.. code-block:: python
+
+   >>> keys = [k for k in mol.keys() if 'lg' in k]
+   >>> temp = np.array([float(k.split('(')[-1].split(')')[0]) for k in keys])
+   >>> part = list(mol[keys][0])
+   >>> plt.scatter(temp,part)
+   >>> plt.xlabel('Temperature (K)')
+   >>> plt.ylabel('Partition Function Value')
+   >>> plt.title('Parititon Fn vs Temp')
+   >>> plt.show()
+
+
+.. figure:: images/docplot_cdms_q.png
+   :scale: 50%
+   :alt: Plot of Partition Function vs Temperature
+
+   The resulting plot from the example above
+
+For non-linear molecules like H2CO, curve fitting methods can be used to
+calculate production rates at different temperatures with the proportionality:
+``a*T**(3./2.)``. Calling the process above for the H2CO molecule (instead of
+for the CO molecule) we can continue to determine the partition function at
+other temperatures using curve fitting models:
+
+.. code-block:: python
+
+   >>> result = CDMS.get_species_table()
+   >>> mol = result[result['TAG'] == 30501] #do not include signs of TAG for this
+   >>> print(mol)
+   >>> from scipy.optimize import curve_fit
+   >>> def f(T, a):
+           return np.log10(a*T**(1.5))
+   >>> keys = [k for k in mol.keys() if 'lg' in k]
+   >>> def tryfloat(x):
+           try:
+               return float(x)
+           except:
+               return np.nan
+   >>> temp = np.array([float(k.split('(')[-1].split(')')[0]) for k in keys])
+   >>> part = np.array([tryfloat(x) for x in mol[keys][0]])
+   >>> param, cov = curve_fit(f, temp[np.isfinite(part)], part[np.isfinite(part)])
+   >>> print(param)
+   # array([0.51865074])
+   >>> x = np.linspace(2.7,500)
+   >>> y = f(x,param[0])
+   >>> plt.scatter(temp,part,c='r')
+   >>> plt.plot(x,y,'k')
+   >>> plt.title('Partition Function vs Temperature')
+   >>> plt.xlabel('Temperature')
+   >>> plt.ylabel('Log10 of Partition Function')
+   >>> plt.show()
+
+
+.. figure:: images/docplot_cdms_fitted.png
+   :scale: 50%
+   :alt: Plot of Partition Function vs Temperature and resulting Curve Fit
+
+   The resulting plot from the example above
+
+
+We can then compare linear interpolation to the fitted interpolation above:
+
+
+.. code-block:: python
+
+   >>> interp_Q = np.interp(x, temp, 10**part)
+   >>> pl.plot(x, (10**y-interp_Q)/10**y)
+   >>> pl.xlabel("Temperature")
+   >>> pl.ylabel("Fractional difference between linear and fitted")
+   
+.. figure:: images/docplot_cdms_fitted_vs_linear.png
```

Comment:
don't know how to do that, actually!  I was just going off the jplspec example

Reply:
there are some good examples to use the `plot` directive, e.g in photutils:
https://raw.githubusercontent.com/astropy/photutils/main/docs/getting_started.rst

or if the code is standalone rather than part  of the flow, then you can include it with `include_source`, too, see e.g. modeling: https://raw.githubusercontent.com/astropy/astropy/main/docs/modeling/index.rst